$OpenBSD: patch-playercode_mloader_c,v 1.1 2013/01/02 15:13:02 stsp Exp $
Fix a NULL dereference triggered by the games/ri-li port.
--- playercode/mloader.c.orig	Mon Dec 31 01:30:02 2012
+++ playercode/mloader.c	Mon Dec 31 01:30:54 2012
@@ -450,10 +450,12 @@ MODULE* Player_LoadGeneric_internal(MREADER *reader,in
 	if (!l->Init || l->Init()) {
 		_mm_rewind(modreader);
 		ok = l->Load(curious);
-		/* propagate inflags=flags for in-module samples */
-		for (t = 0; t < of.numsmp; t++)
-			if (of.samples[t].inflags == 0)
-				of.samples[t].inflags = of.samples[t].flags;
+		if (ok) {
+			/* propagate inflags=flags for in-module samples */
+			for (t = 0; t < of.numsmp; t++)
+				if (of.samples[t].inflags == 0)
+					of.samples[t].inflags = of.samples[t].flags;
+		}
 	} else
 		ok = 0;
 
